### PR TITLE
commit TrainingJobUpdater struct

### PR DESCRIPTION
**Pull request by qizheng09**
This pr commit the `TrainingJobUpdater` struct

### CODE REVIEW

File: go/updater/trainingJobUpdater.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	// kubeCli is standard kubernetes client.
+	kubeCli kubernetes.Interface
+
+	// TrainingJobClient is the client of TrainingJob.
+	trainingJobClient trainingJobClient.Interface
+
+	// Status is the status in memory, update when TrainingJob status changed and update the CRD resource status.
+	status padv1.TrainingJobStatus
+
+	// EventCh is the channel received by Controller, include Modify and Delete.
```
Comment (typhoonzero) [inline go/updater/trainingJobUpdater.go]: > is the channel received by Controller

Do you mean "this channel is shared with the controller to receive TrainingJob events"

File: go/updater/trainingJobUpdater.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+	// When trainingJobEvent is Delete it will delete all resources
+	// The maximum is 1000.
```
Comment (typhoonzero) [inline go/updater/trainingJobUpdater.go]: What maximum here? The capacity or time?

File: go/updater/trainingJobUpdater.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+// TrainingJobUpdater is to manager a specific TrainingJob
+type TrainingJobUpdater struct {
+	// Job is the job the TrainingJob manager.
+	job *padv1.TrainingJob
+
+	// kubeCli is standard kubernetes client.
+	kubeCli kubernetes.Interface
+
+	// TrainingJobClient is the client of TrainingJob.
+	trainingJobClient trainingJobClient.Interface
```
Comment (typhoonzero) [inline go/updater/trainingJobUpdater.go]: What is the difference between `kubeCli` and `trainingJobClient`? Shall we use pointer here?

File: go/updater/trainingJobUpdater.go (hunk 1 above)
Comment (qizheng09) [reply go/updater/trainingJobUpdater.go]: The comments should be changed to "received from Controller". Controller use `EventCh` to notify updater.

File: go/updater/trainingJobUpdater.go (hunk 2 above)
Comment (qizheng09) [reply go/updater/trainingJobUpdater.go]: The maximum is the capacity of the `EventCh`.  For a `TrainingJob`,  1000 may enough.

File: go/updater/trainingJobUpdater.go (hunk 3 above)
Comment (qizheng09) [reply go/updater/trainingJobUpdater.go]: `kubeCli` only have the functions of k8s native resources. We need to use the `trainingJobClient` which @m3ngyang committed in  [#624](https://github.com/PaddlePaddle/cloud/pull/624).

Comment (m3ngyang) [reply go/updater/trainingJobUpdater.go]: > Shall we use pointer here?

Interface is the better choice. In golang, interface is a spec that defines the behavior of a given object and Kubernetes also adopts Interface-based programming.

Comment (typhoonzero) [reply go/updater/trainingJobUpdater.go]: @m3ngyang I see, thanks!

File: go/updater/trainingJobUpdater.go (hunk 1 above)
Comment (typhoonzero) [reply go/updater/trainingJobUpdater.go]: Or maybe like: "EventCh receives event from the controller"?

File: go/updater/trainingJobUpdater.go (hunk 2 above)
Comment (typhoonzero) [reply go/updater/trainingJobUpdater.go]: Okay.

File: go/updater/trainingJobUpdater.go (hunk 1 above)
Comment (qizheng09) [reply go/updater/trainingJobUpdater.go]: Okay~

**Pull request merged**